samples/webserver/index.js

```diff
@@ -29,6 +31,42 @@ onload = function() {
 
   var logToScreen = function(log) {
     logger.textContent += log + "\n";
+    logger.scrollTop = logger.scrollHeight;
+  }
+
+  var destroySocketById = function(socketId) {
+    tcpSocket.disconnect(socketId,function(){
+      tcpSocket.close(socketId);
+    });
+  }
+
+  var closeServerSocket = function() {
+    if(serverSocketId) {
+      tcpServer.close(serverSocketId);
+    }
+
+    tcpServer.onAccept.removeListener(onAccept);
+    tcpSocket.onReceive.removeListener(onReceive);
+  }
+
+  var sendReplyToSocket = function(socketId,buffer,keepAlive) {
+    // verify that socket is still connected before trying to send data
+    tcpSocket.getInfo(socketId,function(socketInfo) {
+      if(socketInfo.connected !== true) {
+        destroySocketById(socketId);
+        return;
+      }
+
+      tcpSocket.setKeepAlive(socketId, keepAlive, function() {
```

Comment:
You may want to check `chrome.runtime.lastError` there also since I happen to have an error on Chrome OS Dev Channel. For info, setting a delay to `1` fixed the raised error.

For info, here's the `ui.LATEST` error I get when running it with no delay:
`[31544:31565:0408/094629:ERROR:tcp_socket_libevent.cc(72)] Failed to set TCP_KEEPIDLE on fd: 704: Invalid argument`
Hopefully @reillyeon will be able to help us there.